packages/mobile/src/tokens/selectors.ts

```diff
@@ -30,21 +32,32 @@ export const tokensListSelector = createSelector(tokensByAddressSelector, (token
   return Object.values(tokens).map((token) => token!)
 })
 
-export const tokensWithBalanceSelector = createSelector(tokensListSelector, (tokens) => {
+type TokenBalanceWithUsdPrice = TokenBalance & {
+  usdPrice: BigNumber
+}
+
+export const tokensWithUsdValueSelector = createSelector(tokensListSelector, (tokens) => {
   return tokens.filter((tokenInfo) =>
-    tokenInfo.balance.multipliedBy(tokenInfo.usdPrice).gt(STABLE_TRANSACTION_MIN_AMOUNT)
-  )
+    tokenInfo.balance.multipliedBy(tokenInfo.usdPrice ?? 0).gt(STABLE_TRANSACTION_MIN_AMOUNT)
+  ) as TokenBalanceWithUsdPrice[]
 })
 
-export const coreTokensSelector = createSelector(tokensListSelector, (tokens) => {
-  return tokens.filter((tokenInfo) => tokenInfo.isCoreToken === true)
+export const tokensWithTokenBalanceSelector = createSelector(tokensListSelector, (tokens) => {
+  return tokens.filter((tokenInfo) => tokenInfo.balance.gt(TOKEN_MIN_AMOUNT))
 })
 
 // Tokens sorted by usd balance (descending)
-export const tokensByUsdBalanceSelector = createSelector(tokensListSelector, (tokensList) => {
-  return tokensList.sort((a, b) =>
-    b.balance.multipliedBy(b.usdPrice).comparedTo(a.balance.multipliedBy(a.usdPrice))
-  )
+export const tokensByUsdBalanceSelector = createSelector(
+  tokensWithUsdValueSelector,
+  (tokensList) => {
+    return tokensList.sort((a, b) =>
+      b.balance.multipliedBy(b.usdPrice).comparedTo(a.balance.multipliedBy(a.usdPrice))
+    )
+  }
+)
```

Comment:
I'm having 2nd thoughts on this. I propose we do this:
```suggestion
// Note: tokens which don't have a usd price will be listed last
export const tokensByUsdBalanceSelector = createSelector(
  tokensListSelector,
  (tokensList) => {
    return tokensList.sort(sortByUsdBalance)
  }
)
```